pages/HomePage.ts

```diff
@@ -0,0 +1,27 @@
+import { expect, Page, Locator } from "@playwright/test";
+import { ProductPage } from './ProductPage';
+import { HeaderFragment } from './HeaderFragment';  
+
+export class HomePage {
+  readonly page: Page;
+  readonly productList: Locator;
+  readonly header: HeaderFragment;
+
+  constructor(page: Page) {
+    this.page = page;
+    this.productList = page.locator('[data-test="product-name"]');
+    this.header = new HeaderFragment(page); 
+  }
+
+  async openProduct(productName: string): Promise<ProductPage> {
+    const product = this.productList.getByText(productName);
+    await expect(product).toBeVisible();
+    await product.click();
+    return new ProductPage(this.page);
```

Comment:
Do we need to return an object of another class? Are you going to use chaining of elements?